src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2954,7 +2990,7 @@ private bool HasUnboxingConversion(TypeSymbol source, TypeSymbol destination, re
                 destination.IsNullableType() &&
                 HasUnboxingConversion(source, destination.GetNullableUnderlyingType(), ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.ValueTypeUnboxing;
```

Comment:
This conversion seems particularly oriented on the fact that destination is a Nullable type.  As such, it feels like a value-type-unboxing, even if source was a type-parameter constrained on 'class'.